x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/AlibabaCloudSearchService.java

```diff
@@ -366,4 +396,85 @@ private void checkAlibabaCloudSearchServiceConfig(Model model, InferenceService
 
     private static final String ALIBABA_CLOUD_SEARCH_SERVICE_CONFIG_INPUT = "input";
     private static final String ALIBABA_CLOUD_SEARCH_SERVICE_CONFIG_QUERY = "query";
+
+    private static class Configuration {
+        public static Map<String, ServiceConfiguration> get() {
```

Comment:
I'm not sure if what I am about to suggest is a good idea, but we could use a singleton to avoid instantiating this object on every request?

```java
private static final Map<String, ServiceConfiguration> configuration;
static {
    var configurationMap = ...
    configuration = Collections.unmodifiableMap(configurationMap);
}
```

```java
private static final LazyInitializable<Map<String, ServiceConfiguration>, ?> configuration = new LazyInitializable<>(() -> {
    var configurationMap = ...
    return Collections.unmodifiableMap(configurationMap);
});
```

Reply:
I like that! Updated in https://github.com/elastic/elasticsearch/pull/114862/commits/1c4550b9958b5de868a552714c858bc2dd32f1ce

I am bubbling up the exception but I could catch and return an empty map, WDYT?